packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -230,18 +230,33 @@ const Separator = ({ separator }: Pick<BreadcrumbsProps, 'separator'>) => (
 const BreadcrumbsItem = React.forwardRef((props, forwardedRef) => {
   const { children, className, ...rest } = props;
 
-  const defaultAs = !!props.href ? Anchor : !!props.onClick ? 'button' : 'span';
+  const resolvedAs = !!props.href ? 'a' : !!props.onClick ? 'button' : 'span';
+
+  const commonProps = {
+    className: cx('iui-breadcrumbs-content', className),
+    ref: forwardedRef,
+    ...rest,
+  };
+
+  if (resolvedAs === 'button') {
+    return (
+      <Button
+        styleType='borderless'
+        {...(commonProps as Partial<React.ComponentProps<typeof Button>>)}
+      >
+        {children}
+      </Button>
+    );
+  }
+
+  if (resolvedAs === 'a') {
+    return <Anchor {...commonProps}>{children}</Anchor>;
```

Comment:
This will not work if the custom `Link` component does not have a `href` prop (which is not an unusual scenario; I've seen `<Link to>`).

Reply:
There are two parts to this issue:

1. `<Breadcrumbs.Item as={Link}>` renders `<Link class="iui-button iui-breadcrumbs-content">` which does not have anchor styling
2. `<Breadcrumbs.Item as="span">` renders `<span class="iui-button iui-breadcrumbs-content">` which has hover styling.

It makes sense to change the rendered element (from `Button` to `Box`) for fixing 2.

```jsx
defaultAs === 'span' ? <Box /> : <Button />
```


But for 1, we could leave the current JS and instead fix the CSS. You can try something like this diff:

<details><summary>breadcrumbs.scss</summary>

```diff
@@ -43,11 +43,15 @@
     text-overflow: ellipsis;
   }
 
-  &:where(a):focus {
+  &:where(:any-link) {
     outline-offset: -1px;
+
+    &:hover {
+      text-decoration: underline;
+    }
   }
 
-  &:where(button) {
+  &:where(button, :any-link) {
     @include base.iui-button;
     @include borderless.iui-button-borderless;
     color: var(--iui-color-text-accent);
```

</details> 

(The `:any-link` selector is basically `a` but actually ensures that `href` is present.)

Reply:
While waiting for your response I took a look at your current code and thought it could be simplified like this:

```tsx
const BreadcrumbsItem = React.forwardRef((props, forwardedRef) => {
  const { as: asProp, ...rest } = props;

  const commonProps = {
    ...rest,
    className: cx('iui-breadcrumbs-content', props.className),
    ref: forwardedRef,
  };

  if (String(asProp) === 'span') {
    return <Box as='span' {...commonProps} />;
  }

  return (
    <Button
      as={(asProp === 'a' ? Anchor : asProp) as any}
      styleType='borderless'
      {...commonProps}
    />
  );
}) as PolymorphicForwardRefComponent<'a'>;
```

Reply:
Ohh, I think this would work if we ask users to explicitly use the `asProp` and not rely on us choose what to render based on the `href`/`onClick`. To confirm, are we hoping to make that change?

When testing in the Remix playground after making the suggested change to `Breadcrumbs.tsx`, all items were rendered in HTML as `<button>`s, except the third item that used the `asProp`:

<details>
<summary>Remix playground</summary>


https://github.com/iTwin/iTwinUI/assets/45748283/2f3c49cc-b8e3-429a-8381-08742f23fec3

```jsx
import type { MetaFunction } from '@remix-run/node';
import { Breadcrumbs } from '@itwin/itwinui-react';
import { Link } from '@remix-run/react';

export const meta: MetaFunction = () => {
  return [
    { title: 'Remix Playground' },
    {
      name: 'description',
      content: 'Playground for testing iTwinUI components in Remix.',
    },
  ];
};

export default function Index() {
  return (
    <>
      <Breadcrumbs>
        <Breadcrumbs.Item key={0} onClick={() => console.log('Button clicked')}>
          button
        </Breadcrumbs.Item>
        <Breadcrumbs.Item key={1} href='/?path=page-refresh'>
          link (as=undefined)
        </Breadcrumbs.Item>
        <Breadcrumbs.Item
          as={Link}
          className='my-special-link'
          key={2}
          to='/?path=no-page-refresh'
        >
          link (as=Link)
        </Breadcrumbs.Item>
        <Breadcrumbs.Item key={3}>span 1</Breadcrumbs.Item>
        <Breadcrumbs.Item key={4}>span 2</Breadcrumbs.Item>
      </Breadcrumbs>
    </>
  );
}

```

</details>

Reply:
I was in progress with testing and fixing the unit tests and so didn't reply after making the relevant commits yesterday. But I believe the latest code work now.

> This will not work if the custom `Link` component does not have a `href` prop (which is not an unusual scenario; I've seen `<Link to>`).

Made changes to render `<Button as={asProp} />` whenever an `asProp` is passed. I also updated the CSS styles like you suggested to add anchor type styling even when rendering using `<Button as={Link} />`.

> There are two parts to this issue:
> 
> 1. `<Breadcrumbs.Item as={Link}>` renders `<Link class="iui-button iui-breadcrumbs-content">` which does not have anchor styling
> 2. `<Breadcrumbs.Item as="span">` renders `<span class="iui-button iui-breadcrumbs-content">` which has hover styling.

I modified the code `Breadcrumbs.Item` code to render a:
* `Button`
  * `as={undefined}` when `onClick` is passed
  * `as={Anchor}` when `href` is passed
  * `as={asProp}` when `as: asProp` is passed
* `Box as={"span"}` in all other cases

As a result:
* even links have the button hover styling, as it is in the current version (v3.11.0). However, with this PR, the anchor styling is properly added when using `as={Link}` since the css has been modified to target `:any-link`.
* `span`s no longer get the button hover styling.

I tested all this logic in NextJS (using `<Link href={} />`) and RemixJS (`<Link to={} />`):
* v3.10.1 (version before this PR): [NextJS](https://stackblitz.com/edit/nextjs-yjodea?file=app%2Fpage.tsx), [RemixJS](https://stackblitz.com/edit/remix-run-remix-6nbtbt?file=README.md)
* v3.9.1 (version before the regression started from #1919): [NextJS](https://stackblitz.com/edit/nextjs-7htile?file=app%2Fpage.tsx), [RemixJS](https://stackblitz.com/edit/remix-run-remix-zfsuvc?file=package.json)
* this PR: [NextJS](https://gist.github.com/rohan-kadkol/ab300ec0772c31c8a46e912ecdb151e0), [RemixJS](https://gist.github.com/rohan-kadkol/139187b0ce222f99536284521e7991ed)

---

All these discussed changes are part of https://github.com/iTwin/iTwinUI/pull/2055/files/6402e0d17323d3d9091a83c0e1fd0fa4e5eb1525..1df2df5e134843ac81c42e78ef5b520e4948b808#diff-8e450321e4016954cde3ba7c086d86f6286650a88b49a9bfc57c39feead6155c.

Reply:
> Ohh, I think this would work if we ask users to explicitly use the `asProp` and not rely on us choose what to render based on the `href`/`onClick`. To confirm, are we hoping to make that change?

Whoops, no. We need to keep the auto behavior (though I think in the future we should try to handle it in `Button` or `ButtonBase` or a different utility component, since it's useful to have it in many places).

Updated code with slight adjustment:

```jsx
const BreadcrumbsItem = React.forwardRef((props, forwardedRef) => {
  const { as: asProp, ...rest } = props;

  const commonProps = {
    ...rest,
    className: cx('iui-breadcrumbs-content', props.className),
    ref: forwardedRef,
  };

  if (String(asProp) === 'span') {
    return <Box as='span' {...commonProps} />;
  }

  return (
    <Button
      as={(asProp === 'a' || !!props.href ? Anchor : asProp) as any}
      styleType='borderless'
      {...commonProps}
    />
  );
}) as PolymorphicForwardRefComponent<'a'>;
BreadcrumbsItem.displayName = 'Breadcrumbs.Item';
```

Reply:
Ohh, I see, sounds good. I flipped the order of the returns to try to simplify the if condition (https://github.com/iTwin/iTwinUI/pull/2055/commits/45dd08d53aa1864aa1e99d5bdfc50dd925e91eb9).

To make it work, I had to add some more logic in the first `if` statement. But this is still lesser code than what was there before.